### PR TITLE
Create TextField and any View

### PR DESCRIPTION
**Pull request by IluyaChief**
Create TextField and any View

### CODE REVIEW

File: TextFieldTask/TextFieldTask/CustomTextFieldStyle.swift [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    
+    
+    var textChanged: ((UITextField?) -> Void)?
+    var returnTapped: ((String?) -> Void)?
+    var focusedForTextField: ((String?) -> Void)?
+    var focusWillBeLost: ((UITextField?) -> Void)?
+    
+    
+    
+
```
Comment (Shroednick) [inline TextFieldTask/TextFieldTask/CustomTextFieldStyle.swift:18]: Delete empty lines

File: TextFieldTask/TextFieldTask/CustomTextFieldStyle.swift [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        labelNumberFunc()
+        textFieldLine()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    private func setupAutoLoyautTextFiled() {
+        addSubview(textField)
```
Comment (Shroednick) [inline TextFieldTask/TextFieldTask/CustomTextFieldStyle.swift:33]: move addSubview to another func like setupSubviews

File: TextFieldTask/TextFieldTask/CustomTextFieldStyle.swift [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+            textField.leftAnchor.constraint(equalTo: self.leftAnchor),
+            textField.heightAnchor.constraint(equalToConstant: 55)
+        ])
+        textField.borderStyle = UITextField.BorderStyle.roundedRect
+        textField.placeholder = "press 6 character"
+        textField.addTarget(self, action: #selector(textChanedEdition), for: .editingChanged)
+        textField.addTarget(self, action: #selector(focusWillLost), for: .editingDidEnd)
+        textField.addTarget(self, action: #selector(tappedReturn), for: .editingDidEndOnExit)
+        textField.addTarget(self, action: #selector(returnFocus), for: .editingDidBegin)
+        textField.addTarget(self, action: #selector(labelNumberFunc), for: .editingChanged)
```
Comment (Shroednick) [inline TextFieldTask/TextFieldTask/CustomTextFieldStyle.swift:48]: change the name of this method

File: TextFieldTask/TextFieldTask/CustomTextFieldStyle.swift [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+            labelTextField.widthAnchor.constraint(equalToConstant: 25),
+            labelTextField.topAnchor.constraint(equalTo: textField.bottomAnchor, constant: -25),
+            labelTextField.rightAnchor.constraint(equalTo: textField.rightAnchor)
+        ])
+        labelTextField.text = String(textField.text!.count)
+        labelTextField.textColor = .red
+    }
+    
+    private func labelOnTextField() {
+        addSubview(labelNumber)
```
Comment (Shroednick) [inline TextFieldTask/TextFieldTask/CustomTextFieldStyle.swift:79]: move add subview to another method like setupSubviews

File: TextFieldTask/TextFieldTask/CustomTextFieldStyle.swift [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+            labelNumber.topAnchor.constraint(equalTo: textField.topAnchor, constant: 0),
+            labelNumber.leftAnchor.constraint(equalTo: textField.leftAnchor)
+        ])
+        labelNumber.isHidden = true
+        labelNumber.textColor = .black
+        labelNumber.text = "press 6 character"
+        labelNumber.font = UIFont.systemFont(ofSize: 15)
+    }
+    private func textFieldLine() {
+        addSubview(textLabelBottom)
```
Comment (Shroednick) [inline TextFieldTask/TextFieldTask/CustomTextFieldStyle.swift:91]: move add subview to another method like setupSubviews

File: TextFieldTask/TextFieldTask/CustomTextFieldStyle.swift [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+        textLabelBottom.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate ([
+            textLabelBottom.topAnchor.constraint(equalTo: textField.bottomAnchor),
+            textLabelBottom.widthAnchor.constraint(equalToConstant: 175),
+            textLabelBottom.heightAnchor.constraint(equalToConstant: 5)
+        ])
+        textLabelBottom.isHidden = true
+    }
+    
+    @objc func textChanedEdition(text: UITextField) -> Bool {
```
Comment (Shroednick) [inline TextFieldTask/TextFieldTask/CustomTextFieldStyle.swift:101]: fix lexical errors

File: TextFieldTask/TextFieldTask/CustomTextFieldStyle.swift [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+        labelNumber.isHidden = false
+            textLabelBottom.isHidden = false
+            textLabelBottom.backgroundColor = .red
+            textField.textColor = .red
+        } else if textLenght > 5 {
+        labelNumber.isHidden = true
+            textLabelBottom.isHidden = false
+            textLabelBottom.backgroundColor = .green
+            textField.textColor = .none
+
```
Comment (Shroednick) [inline TextFieldTask/TextFieldTask/CustomTextFieldStyle.swift:114]: delete empty lines

File: TextFieldTask/TextFieldTask/ViewController.swift [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+class ViewController: UIViewController {
+    let textFieldStyle = CustomTextFieldView()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.addSubview(textFieldStyle)
+        setupAutoLayout()
+        
+        textFieldStyle.focusWillBeLost = { textFieldResult in print("Focus Will Be lost")}
+
```
Comment (Shroednick) [inline TextFieldTask/TextFieldTask/ViewController.swift:12]: delete empty lines

File: TextFieldTask/TextFieldTask/ViewController.swift [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+         
+        textFieldStyle.focusedForTextField = { focused in print ("focus return")}
+    }
+    
+    
+    func setupAutoLayout() {
+        textFieldStyle.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            textFieldStyle.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            textFieldStyle.widthAnchor.constraint(equalToConstant: 200),
```
Comment (Shroednick) [inline TextFieldTask/TextFieldTask/ViewController.swift:25]: move this constants to enum like Consts

File: TextFieldTask/TextFieldTask/CustomTextFieldStyle.swift [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+         return true
+    }
+    
+    @objc func tappedReturn() {
+        returnTapped?("text")
+    }
+    
+    @objc func returnFocus(text: UITextField) ->Bool {
+        focusedForTextField?("text")
+        let textLenght = text.text!.count
```
Comment (Shroednick) [inline TextFieldTask/TextFieldTask/CustomTextFieldStyle.swift:135]: delete force unwrap

File: TextFieldTask/TextFieldTask/CustomTextFieldStyle.swift [hunk 11]
```diff
@@ -0,0 +1,151 @@
...
+            textLabelBottom.backgroundColor = .green
+            textField.textColor = .none
+
+        }
+        return true
+    }
+    
+    @objc func focusWillLost(textField: UITextField) -> Bool {
+        focusWillBeLost?(textField)
+        let textLenght = textField.text!.count
```
Comment (Shroednick) [inline TextFieldTask/TextFieldTask/CustomTextFieldStyle.swift:121]: delete force unwrap

File: TextFieldTask/TextFieldTask/CustomTextFieldStyle.swift [hunk 12]
```diff
@@ -0,0 +1,151 @@
...
+        if textLenght > 0 {
+            textLabelBottom.isHidden = false
+            textLabelBottom.backgroundColor = .white
+        }
+         return true
+    }
+    
+    @objc func tappedReturn() {
+        returnTapped?("text")
```
Comment (Shroednick) [inline TextFieldTask/TextFieldTask/CustomTextFieldStyle.swift:130]: code review

File: TextFieldTask/TextFieldTask/CustomTextFieldStyle.swift [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+    }
+    
+    @objc func returnFocus(text: UITextField) ->Bool {
```
Comment (Shroednick) [inline TextFieldTask/TextFieldTask/CustomTextFieldStyle.swift:133]: create outline

File: TextFieldTask/TextFieldTask/CustomTextFieldStyle.swift [hunk 14]
```diff
@@ -0,0 +1,151 @@
...
+        labelOnTextField()
+        labelNumberFunc()
+        textFieldLine()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    private func setupAutoLoyautTextFiled() {
```
Comment (Shroednick) [inline TextFieldTask/TextFieldTask/CustomTextFieldStyle.swift:32]: setupAutoLoyautTextFiled change to the setupAutoLayout

File: TextFieldTask/TextFieldTask/CustomTextFieldStyle.swift (hunk 3 above)
Comment (IluyaChief) [inline TextFieldTask/TextFieldTask/CustomTextFieldStyle.swift:48]: what it this

**Review by Shroednick — CHANGES_REQUESTED** [review]

**Pull request closed without merging**